File: Leetcode - Java/94_tree_inorderTraversal.java
// https://leetcode.com/problems/binary-tree-inorder-traversal/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> inorder = new ArrayList<Integer>();
        if(root != null) {
            inorder.addAll(inorderTraversal(root.left));
            inorder.add(root.val);
            inorder.addAll(inorderTraversal(root.right));
        }
        return inorder;
    }
}

File: Leetcode-Python-New/94_tree_inorderTraversal.py
# https://leetcode.com/problems/binary-tree-inorder-traversal/

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        def traversal(node: TreeNode) -> None:
            if node is None: 
                return
            traversal(node.left)
            inorder.append(node.val)
            traversal(node.right)
        inorder = []
        traversal(root)
        
        return inorder